### PR TITLE
refactor(editor): Migrate `LogStreaming.store.ts` to composition API

### PR DESCRIPTION
**Pull request by RicardoE105**
## Summary

Title self-explanatory

## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/editor-ui/src/stores/logStreaming.store.ts [hunk 1]
```diff
@@ -32,221 +33,244 @@ export interface DestinationSettingsStore {
...
+	const getAllDestinations = () => {
+		const destinations: MessageEventBusDestinationOptions[] = [];
+		for (const key of Object.keys(items)) {
+			destinations.push(items[key].destination);
+		}
+		return destinations;
+	};
+
+	const clearDestinations = () => {
+		Object.keys(items).forEach((key) => delete items[key]);
```
Comment (tomi) [inline packages/editor-ui/src/stores/logStreaming.store.ts:90]: Why not `items = {}`?

File: packages/editor-ui/src/stores/logStreaming.store.ts [hunk 2]
```diff
@@ -32,221 +33,244 @@ export interface DestinationSettingsStore {
...
-					} as DestinationStoreItem;
+		}
+	};
+
+	const removeDestinationItemTree = (id: string) => {
+		delete items[id];
+	};
+
+	const clearDestinationItemTrees = () => {
+		Object.keys(items).forEach((key) => delete items[key]);
```
Comment (tomi) [inline packages/editor-ui/src/stores/logStreaming.store.ts:154]: Why not `items = {}`?. Also isn't this exactly same as `clearDestinations`? Maybe we can combine the two

File: packages/editor-ui/src/stores/logStreaming.store.ts [hunk 3]
```diff
@@ -32,221 +33,244 @@ export interface DestinationSettingsStore {
...
+	};
+
+	const updateDestination = (destination: MessageEventBusDestinationOptions) => {
+		if (destination.id && items[destination.id]) {
+			items[destination.id].destination = destination;
+			items[destination.id] = { ...items[destination.id] };
```
Comment (tomi) [inline packages/editor-ui/src/stores/logStreaming.store.ts:160]: Is this needed anymore? At least it's different from what the original was, which updated the whole `items` object not just a single key:

```
// to trigger refresh
state.items = { ...state.items };
```

**Review by tomi — COMMENTED** [review]
Looking mainly good. Couple questions to doublecheck

File: packages/editor-ui/src/stores/logStreaming.store.ts [hunk 4]
```diff
@@ -32,221 +33,238 @@ export interface DestinationSettingsStore {
...
+	const getAllDestinations = () => {
+		const destinations: MessageEventBusDestinationOptions[] = [];
+		for (const key of Object.keys(items)) {
+			destinations.push(items[key].destination);
+		}
+		return destinations;
+	};
+
+	const clearDestinations = () => {
+		Object.assign(items, {});
```
Comment (tomi) [inline packages/editor-ui/src/stores/logStreaming.store.ts:90]: This doesn't actually clear anything. It assigns an empty object into items, which results into items staying the same.

```js
> var items = { a: 1, b: 2 }
undefined
> Object.assign(items, {})
{ a: 1, b: 2 }
> items
{ a: 1, b: 2 }
```

**Review by tomi — CHANGES_REQUESTED** [review]

File: packages/editor-ui/src/stores/logStreaming.store.ts (hunk 4 above)
Comment (RicardoE105) [reply packages/editor-ui/src/stores/logStreaming.store.ts:90]: Good catch! 

https://github.com/vuejs/core/issues/1081

Comment (tomi) [reply packages/editor-ui/src/stores/logStreaming.store.ts]: 🤯 that is so confusing syntax and looks like it's wrong. Glad you changed it 👍

**Review by tomi — APPROVED** [review]
🚀

**Comment by janober** [comment]
Got released with [`n8n@1.61.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.61.0)

**Pull request merged**